nl/nl_linux.go

```diff
@@ -627,11 +637,19 @@ done:
 				err = syscall.Errno(-errno)
 
 				unreadData := m.Data[4:]
-				if m.Header.Flags&unix.NLM_F_ACK_TLVS != 0 && len(unreadData) > syscall.SizeofNlMsghdr {
-					// Skip the echoed request message.
-					echoReqH := (*syscall.NlMsghdr)(unsafe.Pointer(&unreadData[0]))
-					unreadData = unreadData[nlmAlignOf(int(echoReqH.Len)):]
 
+				if m.Header.Type == unix.NLMSG_ERROR {
```

Comment:
No need for this check if https://github.com/vishvananda/netlink/pull/1139/files#r2476798656 is the case, I think.